### PR TITLE
feat(content-warehouse): create ruleset sample

### PR DESCRIPTION
**Pull request by chasinandrew**
## Description

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist

- [X] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [X] `pom.xml` parent set to latest `shared-configuration`
- [X] Appropriate changes to README are included in PR
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [X] **Tests** pass:   `mvn clean verify` **required**
- [X] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [X] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [X] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample 
- [X] Please **merge** this PR for me once it is approved

### CODE REVIEW

**Comment by anguillanneuf** [comment]
@chasinandrew i don't have all the context for content-warehouse. do you have a bug like b/271994104 where your team has been properly onboarded? 

cc: @gericdong

**Comment by chasinandrew** [comment]
hey @anguillanneuf I raised a bug for this request b/285617061.

File: content-warehouse/src/main/java/contentwarehouse/v1/CreateRuleSet.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+import com.google.cloud.contentwarehouse.v1.RuleSetServiceSettings;
+import com.google.cloud.resourcemanager.v3.Project;
+import com.google.cloud.resourcemanager.v3.ProjectName;
+import com.google.cloud.resourcemanager.v3.ProjectsClient;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+
+// [START contentwarehouse_createruleset]
```
Comment (anguillanneuf) [inline content-warehouse/src/main/java/contentwarehouse/v1/CreateRuleSet.java:44]: please update sample region tag to match internal sample region tag database. see b/285617061#comment2.

File: content-warehouse/src/main/java/contentwarehouse/v1/CreateRuleSet.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+      // Add Delete Document Action to Action Object 
+      ActionOrBuilder action = Action.newBuilder()
+            .setDeleteDocumentAction((DeleteDocumentAction) deleteDocumentAction).build();
+            
+      // Create rule to add to rule set 
+      RuleOrBuilder rule = Rule.newBuilder()
+          .setTriggerType(TriggerType.ON_CREATE)
+          .setCondition("documentType == 'W9' && STATE =='CA' ")
+          .setActions(0, (Action) action).build();
```
Comment (anguillanneuf) [inline content-warehouse/src/main/java/contentwarehouse/v1/CreateRuleSet.java:83]: use `addActions()` instead.

File: content-warehouse/src/main/java/contentwarehouse/v1/CreateRuleSet.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+      ListRuleSetsRequestOrBuilder listRuleSetsRequest = 
+          ListRuleSetsRequest.newBuilder()
+              .setParent(parent);
+
+      ListRuleSetsPagedResponse listRuleSetsPagedResponse = 
+          ruleSetServiceClient.listRuleSets((ListRuleSetsRequest) listRuleSetsRequest);
+            
+      while (listRuleSetsPagedResponse.getNextPageToken() != null) { 
+        System.out.println("Rule sets:" + listRuleSetsPagedResponse.getPage().toString());
+      }
```
Comment (anguillanneuf) [inline content-warehouse/src/main/java/contentwarehouse/v1/CreateRuleSet.java:111]:
```suggestion
      listRuleSetsPagedResponse.iterateAll().forEach(
          (ruleSet -> System.out.println(ruleSet))
      );
```

File: content-warehouse/src/main/java/contentwarehouse/v1/CreateRuleSet.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+import com.google.cloud.contentwarehouse.v1.RuleSetServiceSettings;
+import com.google.cloud.resourcemanager.v3.Project;
+import com.google.cloud.resourcemanager.v3.ProjectName;
+import com.google.cloud.resourcemanager.v3.ProjectsClient;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+
+// [START contentwarehouse_create_rule_set]
```
Comment (anguillanneuf) [inline content-warehouse/src/main/java/contentwarehouse/v1/CreateRuleSet.java:44]: Move this tag to line 18. We want to display the full sample including imports when we publish this sample.

**Review by anguillanneuf — CHANGES_REQUESTED** [review]
One more nit and you're good to go! Thanks for contributing :-)

**Review by anguillanneuf — APPROVED** [review]

**Review by anguillanneuf — APPROVED** [review]

**Pull request merged**